src/status.rs

```diff
@@ -192,6 +186,264 @@ impl Status {
+        maybe_serialize!(self.cam, "cam");
+        maybe_serialize!(self.feeds, "feeds");
+        maybe_serialize!(self.events, "events");
+        maybe_serialize!(self.radio_show, "radio_show");
+        maybe_serialize!(self.cache, "cache");
+        serialize!(self.issue_report_channels, "issue_report_channels");
+        serialize!(self.state, "state");
+        maybe_serialize!(self.sensors, "sensors");
+        maybe_serialize!(self.ext_versions, "ext_versions");
+        serializer.serialize_struct_end(state)
+    }
+}
+
+
+impl Deserialize for Status {
+    fn deserialize<D: Deserializer>(deserializer: &mut D) -> Result<Self, D::Error> {
+        enum Field {
+            Api,
+            Space,
+            Logo,
+            Url,
+            Location,
+            Contact,
+            Spacefed,
+            Projects,
+            Cam,
+            Feeds,
+            Events,
+            RadioShow,
+            Cache,
+            IssueReportChanels,
+            State,
+            Sensors,
+            ExtVersions,
+        };
+
+        impl Deserialize for Field {
+            fn deserialize<D: Deserializer>(deserializer: &mut D) -> Result<Field, D::Error> {
+                struct FieldVisitor;
+
+                impl Visitor for FieldVisitor {
+                    type Value = Field;
+
+                    fn visit_str<E: SerdeError>(&mut self, value: &str) -> Result<Field, E> {
+                        match value {
+                            "api" => Ok(Field::Api),
+                            "space" => Ok(Field::Space),
+                            "logo" => Ok(Field::Logo),
+                            "url" => Ok(Field::Url),
+                            "location" => Ok(Field::Location),
+                            "contact" => Ok(Field::Contact),
+                            "spacefed" => Ok(Field::Spacefed),
+                            "projects" => Ok(Field::Projects),
+                            "cam" => Ok(Field::Cam),
+                            "feeds" => Ok(Field::Feeds),
+                            "events" => Ok(Field::Events),
+                            "radio_show" => Ok(Field::RadioShow),
+                            "cache" => Ok(Field::Cache),
+                            "issue_report_channels" => Ok(Field::IssueReportChanels),
+                            "state" => Ok(Field::State),
+                            "sensors" => Ok(Field::Sensors),
+                            "ext_versions" => Ok(Field::ExtVersions),
+                            _ => Err(SerdeError::unknown_field(value)),
+                        }
+                    }
+                }
+                deserializer.deserialize_struct_field(FieldVisitor)
+            }
+        }
+
+        struct StatusVisitor;
+
+        impl Visitor for StatusVisitor {
+            type Value = Status;
+
+            fn visit_seq<V: SeqVisitor>(&mut self, mut visitor: V) -> Result<Status, V::Error> {
```

Comment:
Ah, actually there's a default implementation that just throws an error:

https://docs.serde.rs/src/serde/de/mod.rs.html#578-584

Should we get rid of the seq implementation to simplify the code?